src/tsolvers/lasolver/FarkasInterpolator.cc

```diff
@@ -2,692 +2,675 @@
-        return ((it == candidates.end() && color == icolor_t::I_A)
-                || (it == candidates.begin() && color == icolor_t::I_B))
-                ? logic.getTerm_false() : logic.getTerm_true();
-    }
-    std::vector<ItpHelper> helpers;
-    ArithLogic & logic = this->logic;
-    std::transform(candidates.begin(), it, std::back_inserter(helpers),
-                   [&logic](std::pair<PtAsgn, Real> const & expl) {
-                       return ItpHelper{logic, expl.first, expl.second};
-                   });
-    statsHelper.moreThanOneInequality = helpers.size() > 1;
-    using local_terms_t = std::vector<LinearTerm>;
-    // create information about local variables for each inequality
-    std::vector<local_terms_t> ineqs_local_vars;
-    std::vector<ItpHelper> explanations_with_locals;
-    for (const auto & helper : helpers) {
-        local_terms_t local_terms = getLocalTerms(helper, [this, color](PTRef ptr) { return this->isLocalFor(color, ptr); });
-
-        // explanataion with all variables shared form standalone partition
-        if (local_terms.empty()) {
-            statsHelper.standAloneIneq = true;
-            interpolant_inequalities.push_back(helper.negated ? logic.mkNot(helper.explanation) : helper.explanation);
-        }
-        // for explanations with local variables, remember them separately
-        else {
-            ineqs_local_vars.push_back(std::move(local_terms));
-            explanations_with_locals.push_back(helper);
+        struct StatsHelper {
+            bool standAloneIneq = false;
+            bool nonTrivialBasis = false;
+            bool moreThanOneInequality = false;
+        };
+
+    } // namespace
+
+    PTRef FarkasInterpolator::getDecomposedInterpolant(icolor_t color) {
+        assert(color == icolor_t::I_A || color == icolor_t::I_B);
+        bool hasColors = ensureHasColorForAllTerms();
+        if (not hasColors) {
+            throw InternalException(
+                "Error in computation of decomposed Farkas interpolant, colors could not be determined!");
+        }
+        StatsHelper statsHelper;
+        // this will be contain the result, inequalities corresponding to summed up partitions of explanataions (of
+        // given color)
+        std::vector<PTRef> interpolant_inequalities;
+        std::vector<std::pair<PtAsgn, Real>> candidates;
+        assert(explanations.size() == static_cast<int>(explanation_coeffs.size()));
+        for (int i = 0; i < explanations.size(); ++i) {
+            assert(explanation_coeffs[i] > 0);
+            candidates.emplace_back(explanations[i], explanation_coeffs[i]);
+            trace(std::cout << "Explanation " << logic.printTerm(explanations[i].tr) << " with coeff "
+                            << explanation_coeffs[i] << " is negated: " << (explanations[i].sgn == l_False)
+                            << '\n';) bool isA = this->isInPartitionOfColor(icolor_t::I_A, explanations[i].tr);
```

Comment:
Something weird happened here.
The next command is on the same line as the end of `trace`.
Seems clang-format does not work correctly here.

Reply:
Good catch. I changed using of the macro from `trace(statement;)` to `trace(statement);` and now it looks normal.